src/js/shop/components/Checkout/Address.js

```diff
@@ -0,0 +1,197 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { FormattedMessage } from "react-intl";
+import Select from "react-select";
+import { useNavigate } from "react-router-dom";
+
+import { country_list, SUPPORTED_COUNTRIES, EMPTY_ADDRESS } from "./constants";
+import AddressFields from "./AddressFields";
+import PersonalDetailsFields from "./PersonalDetailsFields";
+
+const AddressType = PropTypes.shape({
+    company: PropTypes.string,
+    chamberOfCommerce: PropTypes.string,
+    street: PropTypes.string,
+    number: PropTypes.string,
+    city: PropTypes.string,
+    postalCode: PropTypes.string,
+    country: PropTypes.string,
+});
+
+const CustomerType = PropTypes.shape({
+    firstName: PropTypes.string,
+    lastName: PropTypes.string,
+    email: PropTypes.string,
+    phone: PropTypes.string,
+});
+
+/**
+ *
+ * Address
+ *
+ */
+const Address = ({
+    customer,
+    deliveryAddress,
+    billingAddress,
+    allowSubmit = false,
+    onChange,
+}) => {
+    const [
+        deliveryAddressIsBillingAddress,
+        setDeliveryAddressIsBillingAddress,
+    ] = useState(true);
+    const navigate = useNavigate();
+
+    // TODO this probably needs to send api request to create/modify an order
+    const onSubmit = (event) => {
+        event.preventDefault();
+        navigate("/payment");
+        return;
+    };
+
+    billingAddress =
+        billingAddress ??
+        (!deliveryAddressIsBillingAddress ? EMPTY_ADDRESS : null);
+
+    return (
+        <form onSubmit={onSubmit}>
+            <div className="row">
+                {/*Personal details*/}
+                <div className="col-xs-12 col-md-6">
+                    <h3 className="checkout__title">
+                        <FormattedMessage
+                            description="Checkout address: personal details"
+                            defaultMessage="Personal details"
+                        />
+                    </h3>
+
+                    <PersonalDetailsFields
+                        prefix="customer"
+                        firstName={customer.firstName}
+                        lastName={customer.lastName}
+                        email={customer.email}
+                        phone={customer.phone}
```

Comment:
it's deliberate because the `User`/`UserProfile` models is something I want to get rid of (merge them together). The principle is that I pass the actual data I'm interested in for the relevant components rather than the database structure in the backend. It's a bit more decoupled and flexible.